tests/conftest.py

```diff
@@ -0,0 +1,166 @@
+import pytest
+
+from common import (
+    SCHEMAS_PATH,
+    load_yaml,
+    assert_yaml_header_and_footer,
+    VALID_SCHEMA_FILENAME_RE,
+    split_id,
+    list_refs,
+    path_to_id,
+    path_to_tag,
+    ref_to_id,
+    list_schema_paths,
+    list_latest_schema_paths,
+    DEPRECATED_ID_BASES,
+    METASCHEMA_ID,
+    list_example_ids,
+    list_description_ids,
+)
+
+
+@pytest.fixture(scope="session")
+def schemas():
+    return [load_yaml(p) for p in list_schema_paths(SCHEMAS_PATH)]
+
+
+@pytest.fixture(scope="session")
+def latest_schemas():
+    return [load_yaml(p) for p in list_latest_schema_paths(SCHEMAS_PATH)]
+
+
+@pytest.fixture(scope="session")
+def latest_schema_ids(latest_schemas):
+    result = set()
+    for schema in latest_schemas:
+        if "id" in schema:
+            result.add(schema["id"])
+    return result
+
+
+@pytest.fixture(scope="session")
+def latest_schema_tags(latest_schemas):
+    result = set()
+    for schema in latest_schemas:
+        if "tag" in schema:
+            result.add(schema["tag"])
+        elif "anyOf" in schema:
+            for elem in schema["anyOf"]:
+                if "tag" in elem:
+                    result.add(elem["tag"])
+    return result
+
+
+@pytest.fixture(scope="session")
+def schema_tags(schemas):
+    result = set()
+    for schema in schemas:
+        if "tag" in schema:
+            result.add(schema["tag"])
+        elif "anyOf" in schema:
+            for elem in schema["anyOf"]:
+                if "tag" in elem:
+                    result.add(elem["tag"])
+    return result
+
+
+@pytest.fixture(scope="session")
+def tag_to_schema(schemas):
+    result = {}
+    for schema in schemas:
+        if "tag" in schema:
+            if schema["tag"] not in result:
+                result[schema["tag"]] = []
+            result[schema["tag"]].append(schema)
+    return result
+
+
+@pytest.fixture(scope="session")
+def id_to_schema(schemas):
+    result = {}
+    for schema in schemas:
+        if "id" in schema:
+            if schema["id"] not in result:
+                result[schema["id"]] = []
+            result[schema["id"]].append(schema)
+    return result
+
+
+@pytest.fixture(scope="session")
+def assert_schema_correct(tag_to_schema, id_to_schema):
+    def _assert_schema_correct(path):
```

Comment:
You may want to put a `__traceback__ = False` in this assertion helper so that when there is a failure, you get just the rewritten assertion, not the the traceback of the assertion helper.  This is a handy `pytest` feature.

Reply:
Thanks, I'll do that